src/compiler/abstract_compiler.nit

```diff
@@ -1837,6 +1838,8 @@ redef class MClassType
 			return "char*"
 		else if mclass.name == "NativeArray" then
 			return "val*"
+		else if mclass.name == "UnicodeChar" then
+			return "uint32_t"
```

Comment:
nice :) you managed to pass the ugly and undocumented trap on how the mapping nit <-> C is done internally.